storage/mysql/storage_test.go

```diff
@@ -166,9 +157,83 @@ func nodesAreEqual(lhs []storage.Node, rhs []storage.Node) error {
 	return nil
 }
 
+func openTestDBOrDie() *sql.DB {
+	db, err := OpenDB("test:zaphod@tcp(127.0.0.1:3306)/test")
+	if err != nil {
+		panic(err)
+	}
+	return db
+}
+
+// cleanTestDB deletes all the entries in the database. Only use this with a test database
+func cleanTestDB(db *sql.DB) {
+	// Wipe out anything that was there for this log id
+	for _, table := range allTables {
+		_, err := db.Exec(fmt.Sprintf("DELETE FROM %s", table))
+
+		if err != nil {
+			panic(fmt.Errorf("Failed to delete rows in %s: %s", table, err))
+		}
+	}
+}
+
+func createTestDB(db *sql.DB) {
+	_, err := db.Exec(`REPLACE INTO Trees(TreeId, KeyId, TreeType, LeafHasherType, TreeHasherType)
+					 VALUES(23, "hi", "LOG", "SHA256", "SHA256")`)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// prepareTestTreeDB removes all database contents for the specified log id so tests run in a predictable environment. For obvious reasons this should only be allowed to run against test databases. This method panics if any of the deletions fails to make sure tests can't inadvertently succeed.
+func prepareTestTreeDB(db *sql.DB, treeID int64, t *testing.T) {
```

Comment:
Consider deleting the existing test database and creating a new one for test freshness.